File: backend/src/order/objects/OrderChatUnreadCount.scala
package delivery.order.objects

import delivery.domain.OrderId

final case class OrderChatUnreadCount(
    orderId: OrderId,
    peerRole: OrderChatRole,
    unreadCount: Int,
    latestMessageType: Option[OrderChatMessageType] = None,
    latestContent: Option[String] = None,
    latestCreatedAt: Option[String] = None
)

File: frontend/src/objects/order/OrderChatUnreadCount.ts
import type { OrderChatRole } from '@/objects/order/OrderChatMessage'
import type { OrderId } from '@/objects/shared/ids'

export interface OrderChatUnreadCount {
  orderId: OrderId
  peerRole: OrderChatRole
  unreadCount: number
  latestMessageType?: 'text' | 'image' | string | null
  latestContent?: string | null
  latestCreatedAt?: string | null
}
